apps/igloo-kit-cli/src/utilities/retry.rs

```diff
@@ -0,0 +1,23 @@
+pub async fn retry<E, T, F>(
+    action: impl Fn() -> F,
+    should_retry: fn(u32, &E) -> bool,
```

Comment:
The retry libraries I looked at do not allow checking the error to decide whether to retry.